pkg/plugin/conntrack/_cprog/conntrack.c

```diff
@@ -317,12 +320,15 @@ static __always_inline bool _ct_handle_udp_connection(struct packet *p, struct c
  * @arg reverse_key The reverse key to be used to handle the connection.
  * @arg observation_point The point in the network stack where the packet is observed.
  */
-static __always_inline bool _ct_handle_tcp_connection(struct packet *p, struct ct_v4_key key, struct ct_v4_key reverse_key, __u8 observation_point) {
+static __always_inline bool _ct_handle_tcp_connection(struct packet *p, struct ct_v4_key *key, struct ct_v4_key *reverse_key, __u8 observation_point) {
+    if (!p || !key || !reverse_key) {
+        return false;
+    }
     u8 tcp_handshake = p->flags & (TCP_SYN|TCP_ACK);
     if (tcp_handshake == TCP_SYN) {
         // We have a SYN, we set `is_reply` to false and we provide `key`
         return _ct_create_new_tcp_connection(p, key, observation_point, false);
-    } else if(tcp_handshake == TCP_SYN|TCP_ACK) {
+    } else if(tcp_handshake == (TCP_SYN|TCP_ACK)) {
```

Comment:
Just want to call out what I think is the actual operator precedence problem. My guess is that previously this was being evaluated like `(tcp_handshake == TCP_SYN) | TCP_ACK` which is obviously wrong written that way, but less so given the way it was written previously. The proposed change should correct it to asserting that the handshake is a SYNACK, correct @nddq ?